pkg/rootless/rootless_test.go

```diff
@@ -0,0 +1,84 @@
+// Copyright (c) 2019 Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+package rootless
+
+import (
+	"fmt"
+	"os"
+	"testing"
+
+	//"github.com/sirupsen/logrus"
+	"github.com/stretchr/testify/assert"
+)
+
+func createTestUIDMapFile(input string) error {
+	f, err := os.Create(uidMapPath)
+	if err != nil {
+		return err
+	}
+	defer f.Close()
+
+	_, err = f.WriteString(input)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// TestSetRootlessRootlessUID1000 tests that isRootless is set to
+// true when a host UID is not 0
+func TestSetRootlessRootlessUID1000(t *testing.T) {
+	// by default isRootless should be set to false initially
+	assert.False(t, IsRootless())
+
+	uidMapPathStore := uidMapPath
+	uidMapPath = "/tmp/testUIDMapFile"
+	defer func() {
+		uidMapPath = uidMapPathStore
+		os.RemoveAll(uidMapPath)
+		isRootless = false
+	}()
+
+	mapping := "\t0\t1000\t5555"
+	err := createTestUIDMapFile(mapping)
+	if err != nil {
+		fmt.Printf("Could not create test file: %v", err)
+		os.Exit(1)
```

Comment:
Tests shouldn't really exit - just use assert:

```go
assert.NoError(err)
```

Reply:
even if it technically not a test, but instead a setup for a test?

Reply:
As far as go is concerned, this is a test as the function name starts with `Test`. If you want to perform test setup, either create a `func init()` to do that, or create a [`TestMain()`](https://golang.org/pkg/testing/#hdr-Main) function like:

- https://github.com/kata-containers/runtime/blob/master/virtcontainers/virtcontainers_test.go#L87